### PR TITLE
add adaptive pool and softmax with cross entropy supports different axis, * test = kunlun 

### PR DESCRIPTION
**Pull request by ykkk2333**
### PR types
Others 

### PR changes
OPs

### Describe
add adaptive pool and softmax with cross entropy supports different axis, * test = kunlun

### CODE REVIEW

File: paddle/fluid/operators/softmax_with_cross_entropy_op_xpu.cc [hunk 1]
```diff
@@ -105,6 +103,59 @@ class SoftmaxWithCrossEntropyXPUKernel : public framework::OpKernel<T> {
...
+            loss_data,
+            nullptr,
+            n * d / t,
+            t,
+            ignore_index);
+        PADDLE_ENFORCE_XDNN_SUCCESS(r, "hard_cross_entropy");
+      }
+      return;
+    }
     if (soft_label) {
```
Comment (taixiurong) [inline paddle/fluid/operators/softmax_with_cross_entropy_op_xpu.cc]: 逻辑是否可以简单点，判断是不是rank 是不是 最后一纬度，不是的话，转置，然后调用新接口？？

Comment (runzhech) [reply paddle/fluid/operators/softmax_with_cross_entropy_op_xpu.cc]: 这个hard_cross_entropy是新接口吗？如果是最后一维，且soft_label为false，是不是可以直接调用hard_softmax_with_cross_entropy?

Comment (ykkk2333) [reply paddle/fluid/operators/softmax_with_cross_entropy_op_xpu.cc]: 前向调用已按意见简化修改，反向调用时转置较多，希望还是分情况分开调用接口

Comment (ykkk2333) [reply paddle/fluid/operators/softmax_with_cross_entropy_op_xpu.cc]: > 

调用hard softmax with cross entropy，需要转置softmax的输入，softmax可以支持多种axis输入，实测性能还是分开调用更好，所以还是希望暂时使用分开调用的方式

**Review by runzhech — APPROVED** [review]
lgtm

**Review by taixiurong — APPROVED** [review]

**Review by QingshuChen — APPROVED** [review]
